### PR TITLE
Document ParallelInterleaveDatasetV2 op

### PR DESCRIPTION
**Pull request by frreiss**
This PR adds documentation to the API spec for the `ParallelInterleaveDatasetV2` op in `tf.data`.

I documented all the inputs and attributes in the `REGISTER_OP()` call for the op. I documented how the `num_parallel_calls` attribute affects the degree of parallelism at runtime. I also documented how the Python API creates instances of this op and how the `sloppy` attribute is set in a graph rewrite durning static optimization.

### CODE REVIEW

File: tensorflow/core/api_def/base_api/api_def_ParallelInterleaveDatasetV2.pbtxt [hunk 1]
```diff
@@ -1,13 +1,93 @@
...
+END
+  }
+  attr {
+    name: "output_types"
+  }
+  attr {
+    name: "output_shapes"
+  }
+  attr {
+    name: "sloppy"
```
Comment (jsimsa) [inline tensorflow/core/api_def/base_api/api_def_ParallelInterleaveDatasetV2.pbtxt]: I don't think this attribute is present in the ParallelInterleaveDatasetV2.

File: tensorflow/core/api_def/base_api/api_def_ParallelInterleaveDatasetV2.pbtxt [hunk 2]
```diff
@@ -1,13 +1,93 @@
...
   summary: "Creates a dataset that applies `f` to the outputs of `input_dataset`."
+  description: <<END
+The resulting dataset is similar to the `InterleaveDataset`, except that the
+dataset will fetch records from the interleaved datasets in parallel.
+
+The `tf.data` Python API creates instances of this op from
+`Dataset.interleave()` when the `num_parallel_calls` parameter of that method
+is set to any value other than `None`.
+
+!! WARNING !! If the `sloppy` parameter is set to `True`, the operation of this
```
Comment (jsimsa) [inline tensorflow/core/api_def/base_api/api_def_ParallelInterleaveDatasetV2.pbtxt]: Remove any mention of sloppy from the documentation as the op does not have the attribute.

File: tensorflow/core/api_def/base_api/api_def_ParallelInterleaveDatasetV2.pbtxt [hunk 3]
```diff
@@ -81,13 +71,8 @@ The `tf.data` Python API creates instances of this op from
...
 
-!! WARNING !! If the `sloppy` parameter is set to `True`, the operation of this
-dataset will not be deterministic!
-
-The `sloppy` parameter of this op always starts out set to `False` when the
-Python API creates instances of this op. If the `make_sloppy` static
-optimization pass runs, that optimization will set the `sloppy` parameter to
-`True`. Whether this optimization is performed is determined by the
-`experimental_deterministic` parameter of `tf.data.Options`.
+If the `experimental_deterministic` parameter of `tf.data.Options` is set to
```
Comment (jsimsa) [inline tensorflow/core/api_def/base_api/api_def_ParallelInterleaveDatasetV2.pbtxt]: As the default of that argument is `True`, I would phrase this the other way around, pointing out that in order to avoid head of line blocking, one can set the `experimental_deterministic` parameter to `False` (which will can improve performance at the expense of non-determinism).

File: tensorflow/core/api_def/base_api/api_def_ParallelInterleaveDatasetV2.pbtxt (hunk 1 above)
Comment (frreiss) [reply tensorflow/core/api_def/base_api/api_def_ParallelInterleaveDatasetV2.pbtxt]: Commit a34c748 removed this parameter.

File: tensorflow/core/api_def/base_api/api_def_ParallelInterleaveDatasetV2.pbtxt (hunk 2 above)
Comment (frreiss) [reply tensorflow/core/api_def/base_api/api_def_ParallelInterleaveDatasetV2.pbtxt]: Commit a34c748 removed mentions of the `sloppy` parameter from the description field.

File: tensorflow/core/api_def/base_api/api_def_ParallelInterleaveDatasetV2.pbtxt (hunk 3 above)
Comment (frreiss) [reply tensorflow/core/api_def/base_api/api_def_ParallelInterleaveDatasetV2.pbtxt]: Thanks for pointing that out! I've updated the documentation for deterministic behavior of this op with a more complete explanation of the `experimental_deterministic` parameter in commit 9dbd330.

File: tensorflow/core/api_def/base_api/api_def_ParallelInterleaveDatasetV2.pbtxt [hunk 4]
```diff
@@ -1,6 +1,47 @@
...
+    name: "block_length"
+    description: <<END
+Number of elements at a time to produce from each interleaved invocation of a
+dataset returned by `f`.
+END
+  }
+  in_arg {
+    name: "num_parallel_calls"
+    description: <<END
+If this input is >=1, it bounds the number of datasets from which this instance
```
Comment (jsimsa) [inline tensorflow/core/api_def/base_api/api_def_ParallelInterleaveDatasetV2.pbtxt]: API documentation should not talk about implementation. It should described the behavior. From that perspective, I would reword this to:

Determines the number of threads that should be used for fetching data from input datasets in parallel. The Python API `tf.data.experimental.AUTOTUNE` constant can be used to indicate that the level of parallelism should be autotuned.

**Review by jsimsa — APPROVED** [review]
